### PR TITLE
New S3 plugin

### PR DESCRIPTION
**Pull request by soulgalore**
Skip the old and outdated S3 plugin and use our own instead where you can configure
almost everything for your S3 instance/upload.

### CODE REVIEW

File: lib/plugins/s3/index.js [hunk 1]
```diff
@@ -1,109 +1,128 @@
...
+        resolve();
+      }
+    };
+    const options = { partSize: 10 * 1024 * 1024, queueSize: 1 };
+    const params = {
+      Body: stream,
+      Bucket: s3Options.bucketname,
+      ContentType: contentType,
+      Prefix: s3Options.path || prefix,
+      Key: prefix + file.replace(baseDir, ''), // TODO is this correct if we set the path?
```
Comment (tobli) [inline lib/plugins/s3/index.js]: I think the key should be:
```
// path of the file inside of the dir
const subPath = path.relative(baseDir, file);
let key;
if (s3Options.path) {
  key = path.join(s3Options.path, subPath);
} else {
  key = path.join(prefix, subPath)
}
```

Comment (tobli) [reply lib/plugins/s3/index.js]: or ```const key = path.join(s3Options.path || prefix, subPath)```

File: lib/plugins/s3/index.js [hunk 2]
```diff
@@ -1,109 +1,128 @@
...
+      } else {
+        resolve();
+      }
+    };
+    const options = { partSize: 10 * 1024 * 1024, queueSize: 1 };
+    const params = {
+      Body: stream,
+      Bucket: s3Options.bucketname,
+      ContentType: contentType,
+      Prefix: s3Options.path || prefix,
```
[the remark this answers is not in the document]
Comment (soulgalore) [reply lib/plugins/s3/index.js]: Hmm this is what we used in the old plugin, let me dig into it.

File: lib/plugins/s3/index.js (hunk 1 above)
Comment (soulgalore) [reply lib/plugins/s3/index.js]: Need to test it.

File: lib/plugins/s3/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (soulgalore) [reply lib/plugins/s3/index.js]: You are right, removed it.

File: lib/plugins/s3/index.js (hunk 1 above)
Comment (soulgalore) [reply lib/plugins/s3/index.js]: Fixed.

**Pull request merged**